### PR TITLE
[docs] - Apply style guide to Asset versioning and caching guide

### PR DESCRIPTION
**Pull request by erinkcochran87**
## Summary & Motivation

This PR applies our style guide standards to the Asset versioning and caching guide.

---

## How I Tested These Changes

👀

### CODE REVIEW

File: docs/content/guides/dagster/asset-versioning-and-caching.mdx [hunk 1]
```diff
@@ -52,11 +65,13 @@ width={4064}
...
+The code version shown is a copy of the run ID for the run that generated this materialization. Because `a_number` has no user-defined `code_version`, Dagster assumes a different code version on every run, which it represents with the run ID.
 
-If we materialize the asset again, you'll notice that both the code version and data version change. The code version becomes the ID of the new run and the data version a hash of the new code version.
+The `data_version` is also generated by Dagster. This is a hash of the code version together with the data versions of any inputs. Since `a_number` has no inputs, in this case, the data version is a hash of the code version only.
 
-We can improve this situation by setting an explicit code version. Let's add a `code_version` on our asset:
+If you materialize the asset again, you'll notice that both the code version and data version change. The code version becomes the ID of the new run and the data version, a hash of the new code version.
```
Comment (smackesey) [inline docs/content/guides/dagster/asset-versioning-and-caching.mdx:72]: >and the data version, a hash of the new code version.

"," -> "becomes" ?

File: docs/content/guides/dagster/asset-versioning-and-caching.mdx [hunk 2]
```diff
@@ -120,9 +135,11 @@ def multiplied_number(versioned_number):
     return versioned_number * 2
 ```
 
-Reload the location in dagit and you'll notice that `multiplied_number` is marked as "Never Materalized". Now click on the "Materialize" disclosure triangle and then "Materialize stale and missing" (the plain "Materialize" button ignores versioning) to materialize.
+In the Dagster UI, click **Reload definitions**. The `multipled_number` asset will be marked as **Never materialized**.
+
+Next, click the toggle to the right side of the **Materialize** button to display the **Materialize stale and missing** option. As the **Materialize** button ignores versioning, we need this option to ensure the `multipled_number` asset is properly materialized.
 
-Note that in the created run, _only_ the step associated with `multiplied_number` is run. The system knows that `versioned_number` is up to date and therefore can safely skip that computation. You can see this on the details page for the run:
+**Note**: In the created run, only the step associated with `multiplied_number` is run. The system knows that `versioned_number` is up to date and therefore can safely skip that computation. You can see this on the details page for the run:
```
Comment (smackesey) [inline docs/content/guides/dagster/asset-versioning-and-caching.mdx:142]: I think this is part of the main flow of text so would remove the "**Note**" here.

File: docs/content/guides/dagster/asset-versioning-and-caching.mdx [hunk 3]
```diff
@@ -156,17 +173,17 @@ width={2661}
...
-Click "Materialize stale and missing" to get both assets up-to-date again.
+Click **Materialize stale and missing** to get both assets up-to-date again.
 
 ---
 
-## Step Three: Computing your own data versions
+## Step three: Computing your own data versions
 
-A data version is like a fingerprint for the value that an asset represents, i.e. the output of its materialization function. We therefore want our data versions to correspond 1-1 to the possible return values of a materialization function. Dagster auto-generates data versions by hashing the code version together with input data versions. This satisfies the above criterion in many cases, but sometimes a different approach is necessary.
+A data version is like a fingerprint for the value that an asset represents, i.e. the output of its materialization function. Therefore, we want our data versions to correspond o the possible return values of a materialization function. Dagster auto-generates data versions by hashing the code version together with input data versions. This satisfies the above criterion in many cases, but sometimes a different approach is necessary.
```
Comment (smackesey) [inline docs/content/guides/dagster/asset-versioning-and-caching.mdx:182]: >to correspond o the possible return values

Typo. Not sure what you were intending here but we need to retain the 1-1 concept-- maybe "one-to-one"?

File: docs/content/guides/dagster/asset-versioning-and-caching.mdx [hunk 4]
```diff
@@ -218,25 +237,25 @@ width={2662}
...
 
-Note that if `versioned_number` had used a Dagster-generated data versions, `multiplied_number` would be stale. The data version of `versioned_number` would have changed due to its updated code version, despite the fact that the returned value did not change. `multiplied_number` would then register as stale based off of the updated data version of `versioned_number`.
+**Note**: If `versioned_number` had used a Dagster-generated data version, `multiplied_number` would be stale. The data version of `versioned_number` would have changed due to its updated code version, despite the fact that the returned value did not change. `multiplied_number` would then register as stale based on the updated data version of `versioned_number`.
```
Comment (smackesey) [inline docs/content/guides/dagster/asset-versioning-and-caching.mdx:242]: Per above comment I would also drop `**Note** here

File: docs/content/guides/dagster/asset-versioning-and-caching.mdx [hunk 5]
```diff
@@ -218,25 +237,25 @@ width={2662}
...
 
 ---
 
-## Step Four: Staleness with Source Assets
+## Step four: Staleness with source assets
 
-In the real world, data pipelines depend on external upstream data. So far in this tutorial, we haven't used any external data-- we've been substituting hardcoded data in the asset at the root of our graph, and using a code version as a stand-in for the version of that data. We can do better than this.
+In the real world, data pipelines depend on external upstream data. So far in this guide, we haven't used any external data: we've been substituting hardcoded data in the asset at the root of our graph and using a code version as a stand-in for the version of that data. We can do better than this.
```
Comment (smackesey) [inline docs/content/guides/dagster/asset-versioning-and-caching.mdx:248]: >haven't used any external data: we've been

IMO ":" doesn't fit here-- were you thinking ";" instead?

Alternatively:

>external data. Instead, we've been....

**Review by smackesey — APPROVED** [review]
A few inline comments but otherwise looks good, thanks for cleaning up the "we"/"you" chaos

File: docs/content/guides/dagster/asset-versioning-and-caching.mdx (hunk 5 above)
Comment (erinkcochran87) [reply docs/content/guides/dagster/asset-versioning-and-caching.mdx:248]: Yes; ty

**Pull request merged**